Src/FluentAssertions/Types/PropertyInfoSelector.cs

```diff
@@ -18,18 +18,23 @@ public class PropertyInfoSelector : IEnumerable<PropertyInfo>
         /// Initializes a new instance of the <see cref="PropertyInfoSelector"/> class.
         /// </summary>
         /// <param name="type">The type from which to select properties.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="type"/> is <c>null</c>.</exception>
         public PropertyInfoSelector(Type type)
             : this(new[] { type })
         {
+            Guard.ThrowIfArgumentIsNull(type, nameof(type));
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PropertyInfoSelector"/> class.
         /// </summary>
         /// <param name="types">The types from which to select properties.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="types"/> is <c>null</c>.</exception>
         public PropertyInfoSelector(IEnumerable<Type> types)
         {
-            selectedProperties = types.SelectMany(t => t
+            Guard.ThrowIfArgumentIsNull(types, nameof(types));
+
+            selectedProperties = types.Where(t => t is not null).SelectMany(t => t
```

Comment:
Is the `Where(t => t is not null)` intended to defensively guard against calling the other ctor with a null type?
If yes, then that constructor could instead be re-written to
```cs
public PropertyInfoSelector(Type type)
    : this(new[] { type ?? throw new ArgumentNullException(nameof(type)) })
```

Reply:
No, it was intended to be defensive against

```csharp
new PropertyInfoSelector(new { typeof(int), null, typeof(string) });
```

Reply:
I see...
I don't think we should filter off invalid values like in that example.

As a user I would rather have an Exception thrown in my face here than FA hiding my bug.
I've spent way to many hours in a debugger hunting nulls stemming from methods returning null when getting invalid data.

Reply:
Made it more restrictive